src/planwise/client/current_project/handlers.cljs

```diff
@@ -219,3 +262,101 @@
  in-current-project
  (fn [db [_ new-zoom]]
    (assoc-in db [:map-view :zoom] new-zoom)))
+
+;; ---------------------------------------------------------------------------
+;; Sharing handlers
+
+(register-handler
+ :current-project/open-share-dialog
+ in-current-project
+ (fn [db [_]]
+   ; TODO: Fire a request to update the project shares when opening the dialog to ensure it is up to date
+   ; What happens if the user deletes an item from the list before the request returns? In that case,
+   ; the item that was just deleted will re-appear.
+   (assoc db :view-state :share-dialog)))
+
+(register-handler
+ :current-project/close-share-dialog
+ in-current-project
+ (fn [db [_]]
+   (assoc db :view-state :project)))
+
+(register-handler
+ :current-project/reset-share-token
+ in-current-project
+ (fn [db [_]]
+   (let [id (db/project-id db)]
+     (api/reset-share-token id :current-project/share-token-loaded)
+     (assoc-in db [:sharing :token-state] :reloading))))
+
+(register-handler
+ :current-project/share-token-loaded
+ in-current-project
+ (fn [db [_ {token :token}]]
+   (-> db
+     (assoc-in [:project-data :share-token] token)
+     (assoc-in [:sharing :token-state] nil))))
+
+(defn- remove-project-share
+  [coll user-id]
+  (remove #(= user-id (:user-id %)) coll))
+
+(register-handler
+ :current-project/search-shares
+ in-current-project
+ (fn [db [_ string]]
+   (assoc-in db [:sharing :shares-search-string] string)))
+
+(register-handler
+ :current-project/delete-share
+ in-current-project
+ (fn [db [_ user-id]]
+   (let [project-id (db/project-id db)]
+     (api/delete-share project-id user-id :current-project/share-deleted)
+     (update db :shares asdf/swap! remove-by :user-id user-id))))
+
+(register-handler
+ :current-project/share-deleted
+ in-current-project
+ (fn [db [_ {:keys [user-id project-id]}]]
+   ; TODO: Should we reload the list if accessed when invalidated?
+   ; If we do so, we risk loading a stale list if:
+   ; 1- The user an item from the list
+   ; 2- We optimistically remove it and issue the delete request
+   ; 3- The request returns from the server and we invalidate the list
+   ; 4- The invalidate triggers a full reload of the list
+   ; 5- The user deletes another item, which is optimistically deleted
+   ; 6- The full reload of the list resolves, adding the item just deleted by the user
+   ; Should we cancel the previous request to reload the shares list?
+   ; Or simply ignore the invalid status of the list?
+   (update db :shares asdf/invalidate! remove-by :user-id user-id)))
```

Comment:
Agreed. Removed this in 1dd61096d1eacdef4db4a155fd519a713f98291f, in favour of just reloading the list of shares when opening the share dialog.